http/tests/canvas/ctx.2d-canvas-style-no-document-leak.html is flaky failure

rreno opened the pull request:
#### 148e56013b4570bf1648002252e2282963b39ab7
<pre>
http/tests/canvas/ctx.2d-canvas-style-no-document-leak.html is flaky failure
<a href="https://bugs.webkit.org/show_bug.cgi?id=277234">https://bugs.webkit.org/show_bug.cgi?id=277234</a>
<a href="https://rdar.apple.com/132725257">rdar://132725257</a>

Reviewed by Fujii Hironori.

The 2D canvas style leak tests were not robust to the conservative
nature of the JSC garbage collector. This changes the test to use the
document-leak-test.js helper which will allow us to test against a
number of iframes to maximize the chances we will see a Document object
get collected and reduce flakiness of the test.

* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-color-no-document-leak-expected.txt: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-color-no-document-leak.html: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-gradient-no-document-leak-expected.txt: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-gradient-no-document-leak.html: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-no-document-leak-expected.txt: Removed.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-no-document-leak.html: Removed.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-pattern-no-document-leak-expected.txt: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-pattern-no-document-leak.html: Added.

Canonical link: <a href="https://commits.webkit.org/281523@main">https://commits.webkit.org/281523@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/375d6d043087ba42dee297f6768e41def522b4ae

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60130 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39478 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12682 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/64048 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10660 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47150 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10906 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48711 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7445 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62161 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36808 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52087 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29553 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33513 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9321 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9580 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55431 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9599 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65780 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/4060 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9496 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56066 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4078 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52065 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56220 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13343 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3379 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35291 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36373 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37461 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36118 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

fujii reviewed (approved):
Thank you very much!

rreno committed:
http/tests/canvas/ctx.2d-canvas-style-no-document-leak.html is flaky failure
https://bugs.webkit.org/show_bug.cgi?id=277234
rdar://132725257

Reviewed by Fujii Hironori.

The 2D canvas style leak tests were not robust to the conservative
nature of the JSC garbage collector. This changes the test to use the
document-leak-test.js helper which will allow us to test against a
number of iframes to maximize the chances we will see a Document object
get collected and reduce flakiness of the test.

* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-color-no-document-leak-expected.txt: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-color-no-document-leak.html: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-gradient-no-document-leak-expected.txt: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-gradient-no-document-leak.html: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-no-document-leak-expected.txt: Removed.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-no-document-leak.html: Removed.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-pattern-no-document-leak-expected.txt: Added.
* LayoutTests/http/tests/canvas/ctx.2d-canvas-style-pattern-no-document-leak.html: Added.

Canonical link: https://commits.webkit.org/281523@main
